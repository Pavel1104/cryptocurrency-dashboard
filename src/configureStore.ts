import { createStore, compose, applyMiddleware, GenericStoreEnhancer } from 'redux';
import createSagaMiddleware, { SagaMiddleware } from 'redux-saga';
import persistState, { mergePersistedState } from 'redux-localstorage';
import adapter from 'redux-localstorage/lib/adapters/localStorage';
import filter from 'redux-localstorage-filter';

import { rootReducer, IRootState, Reducer } from '@src/redux_';
import { rootSaga } from '@src/sagas';

// const composeEnhancers = (
  //   process.env.NODE_ENV !== 'production' &&
  //       window &&
  //       window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
  // ) || compose;

// https://github.com/zalmoxisus/redux-devtools-extension/issues/430
const composeEnhancers = compose;

const STORAGE_VERSION = 'v1';

const configureStore = (initialState?: IRootState) => {
  const sagaMiddleware = createSagaMiddleware();

  const reducer = compose<Reducer>(
    mergePersistedState(),
  )(rootReducer);

  const storage = compose(
    filter([
      'currencies.selected',
      'currencies.target',
      'layout.current',
    ]),
  )(adapter(window.localStorage));

  const enhancer = composeEnhancers(
    applyMiddleware(sagaMiddleware),
    persistState(storage, `cryptocurrency-dashboard-${STORAGE_VERSION}`),
  );

  const store = createStore(
    reducer,
    initialState!,
    enhancer as GenericStoreEnhancer,
  );

  sagaMiddleware.run(rootSaga);

  return store as any;
};

export default configureStore;
