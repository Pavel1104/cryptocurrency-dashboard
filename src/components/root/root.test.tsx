import React from 'react';
import { shallow } from 'enzyme';

import { Root } from '.';
import configureStore from '@src/configureStore';

it('Renders root', function () {
  const store = configureStore();

  const wrapper = shallow(
    <Root store={store} />
  );

  expect(wrapper).toMatchSnapshot();
});
