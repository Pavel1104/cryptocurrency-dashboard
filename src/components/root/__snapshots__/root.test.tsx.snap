// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders root 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(withStyles(AppRaw)) />
</Provider>
`;
